astropy/io/fits/tests/test_util.py

```diff
@@ -26,3 +37,109 @@ def test():
                     'KeyboardInterrupt ignored until test is complete!')
 
         pytest.raises(KeyboardInterrupt, test)
+
+
+class TestUtilMode(FitsTestCase):
+    """
+    The high-level tests are partially covered by
+    test_core.TestConvenienceFunctions.test_fileobj_mode_guessing
+    but added some low-level tests as well.
+
+    TODO: These are regression tests for
```

Comment:
Because first I had no regression test for the workaround fail and made this TODO, then I inserted the regression test but after @mhvk said I should delete the "unnecessary" functions I deleted the test again. Long story short: I'll delete the TODO. I thought I caught all of these in the recent cleanup commit. Thanks